### DISCUSSION TITLE
I'm trying to run @QuarkusTest in bazel tests, but PgPool proxy injection fails, any idea where to go from here?

### DISCUSSION THREAD

**Discussion by olev-mutso** [quarkusio/quarkus#32772] [Q&A] [answered]
I'm running @QuarkusTest that connects to postgresql db and makes couple of CRUD operations. This works fine in normal maven environment. 

Problem comes in when I'm trying to run tests via bazel, junit5 runner. I know that bazel is not supported by quarkus so this is just an experiment. 

I'm able to get the to the testcontains part, where postgresql instance is started and proxy bean for PgPool is created.
When quarkus test environment tries to inject the proxy bean into the test, it ends up crashing with:

`Caused by: java.lang.IllegalArgumentException: Can not set io.vertx.mutiny.pgclient.PgPool field io.digiexpress.client.tests.support.TestCase.pgPool to io.vertx.mutiny.pgclient.PgPool_2d3061477c44a6b700dc3f9c4fe453a846636193_Synthetic_ClientProxy
	at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:167)
	at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:171)
	at java.base/jdk.internal.reflect.UnsafeObjectFieldAccessorImpl.set(UnsafeObjectFieldAccessorImpl.java:81)
	at java.base/java.lang.reflect.Field.set(Field.java:793)
	at io.quarkus.arc.impl.Reflections.writeField(Reflections.java:155)
	... 75 more
`

Does anyone have any ideas where to go from here?

[a comment by an automated account is not shown]

**Comment by manovotn** [comment, marked as answer]
I have no experience with Bazel, so take this with a bucket of salt, but looking at the exception, `IllegalArgumentException` should happen when you try to set a field value that's not assignable to the field type.
Now, from a quick look at Quarkus code, I assume the bean in question is [this synth. bean](https://github.com/quarkusio/quarkus/blob/main/extensions/reactive-pg-client/deployment/src/main/java/io/quarkus/reactive/pg/client/deployment/ReactivePgClientProcessor.java#L219-L224); or rather it's client proxy since the bean is normal scoped. Either way, the resulting object should have the `PgPool` type.
I'd probably start by debugging `io.quarkus.arc.impl.Reflections#writeField(...)` to try and spot what exactly is being assigned and why it causes IAE.

As a side note - Quarkus does its fair bit of class loader magic ([as can be seen in this file](https://github.com/quarkusio/quarkus/blob/main/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java)) while setting up a test using `@QuarkusTest`. Some of it might interfere with your experiment as well.

**Reply by olev-mutso** [reply]
Thank you for the quick response. The synth bean is created by the quarkus extension in deployment artifact. Does this mean that I need to have [annotation processor](https://github.com/quarkusio/quarkus/blob/main/core/processor/src/main/java/io/quarkus/annotation/processor/ExtensionAnnotationProcessor.java) defined in my build?

**Reply by mkouba** [reply]
> Does this mean that I need to have [annotation processor](https://github.com/quarkusio/quarkus/blob/main/core/processor/src/main/java/io/quarkus/annotation/processor/ExtensionAnnotationProcessor.java) defined in my build?

Nope. Quarkus does not use an annotation processor when building your app.

> As a side note - Quarkus does its fair bit of class loader magic ([as can be seen in this file](https://github.com/quarkusio/quarkus/blob/main/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java)) while setting up a test using @QuarkusTest. Some of it might interfere with your experiment as well.

Yes, a class loading issue would be my guess...

**Reply by Ladicek** [reply]
I'd also say there's a classloading issue. Clearly `io.vertx.mutiny.pgclient.PgPool_2d3061477c44a6b700dc3f9c4fe453a846636193_Synthetic_ClientProxy` is a subclass of `io.vertx.mutiny.pgclient.PgPool`, so if the JVM says it isn't, it must be because one comes from a different classloader than the other.
